File: sdk/js-sdk/src/api/tournament.ts
import { AxiosInstance } from "axios";

export class TournamentAPI {

    constructor(
        private client: AxiosInstance,
    ) {}

    async list() {

        return (

            await this.client.get("/tournament")

        ).data;

    }

    async create(payload: any) {

        return (

            await this.client.post(

                "/tournament",

                payload,

            )

        ).data;

    }

}

File: sdk/skillbattle/api/tournament.py
class TournamentAPI:

    def __init__(

        self,

        client,

    ):

        self.client = client

    def list(self):

        return self.client.get(

            "/tournament",

        ).json()

    def get(

        self,

        tournament_id,

    ):

        return self.client.get(

            f"/tournament/{tournament_id}",

        ).json()

    def create(

        self,

        payload,

    ):

        return self.client.post(

            "/tournament",

            json=payload,

        ).json()